src/basic/siphash24.h

```diff
@@ -25,6 +25,13 @@ static inline void siphash24_compress_boolean(bool in, struct siphash *state) {
         siphash24_compress(&i, sizeof i, state);
 }
 
+static inline void siphash24_compress_string(const char *in, struct siphash *state) {
+        if (!in)
+                return;
+
+        siphash24_compress(in, strlen(in), state);
```

Comment:
I'm wondering if this should be `strlen() + 1` so that the result for NULL and "" is different.